source/common/common/token_bucket_impl.cc

```diff
@@ -103,4 +104,17 @@ double AtomicTokenBucketImpl::timeNowInSeconds() const {
   return std::chrono::duration<double>(time_source_.monotonicTime().time_since_epoch()).count();
 }
 
+std::chrono::milliseconds AtomicTokenBucketImpl::nextTokenAvailable() const {
+  // If there are tokens available, return immediately.
+  if (remainingTokens() >= 1) {
+    return std::chrono::milliseconds(0);
+  }
+
+  // Calculate time since the last fill.
+  double current_time = timeNowInSeconds();
+  double last_time = time_in_seconds_.load();
+  return std::chrono::milliseconds(
+      static_cast<uint64_t>(((1 / fill_rate_ - (current_time - last_time)) * 1000)));
+}
```

Comment:
Although it's will be a very very very trival case, but it's possible when we check `remainingTokens()`, the remaining tokens are less then 1. Then, when we calculate the next available token, time passes, the `1 / fill_rate_ - (current_time - last_time)` may return a minus value.